content/kapp-controller/docs/latest/debugging.md

```diff
@@ -0,0 +1,113 @@
+
+In cases where the error message does not originate from kapp-controller (e.g. a failed fetch event for a git repository), 
+the stderr from the underlying tool (i.e. vendir, imgpkg, kbld, ytt, kapp, and helm) is shown in the App's status. 
+
+In the App status, there is a field called `usefulErrorMessage` that displays the stderr for a failure during App reconciliation.
+
+This `usefulErrorMessage` field can be found by running `kubectl get apps/simple-app -o=jsonpath={.status.usefulErrorMessage} -n namespace`. 
+The kubectl command will return the stderr output from the App status to help you further understand the reason for the App failure.
+
+The `usefulErrorMessage` can be helpful in pointing out where errors occurred from inputs in the App spec and also 
+pinpoint the resource that caused a deployment failure. However, Apps will not surface errors of resources it is 
+deploying to Kubernetes and further debugging of resources deployed by an App may be needed.
+
+### Debugging InstalledPackage CRs
+
+Failures for InstalledPackages can be directly viewed via the InstalledPackage's status via the `usefulErrorMessage` property of
+the status. This `usefulErrorMessage` property comes from an App CR that is created as a result of creating an InstalledPackage. More 
+information on interpretting the error message from `usefulErrorMessage` can be found under the [Debugging App CRs](#debugging-app-crs). 
+The underlying App CR will have the same name as the InstalledPackage that you create.
+
+In addition to understanding the error from the underlying App CR, it can also be useful the inspect the Package definition to 
+see if there are issues for the Package being deployed by the InstalledPackage.
+
+The name of the Package should be a combination of the Package's `publicName` and `version`. You can view the Package details 
+by taking the `.spec.packageRef.publicName` and `.spec.packageRef.version` values from the InstalledPackage and run the following:
+
+```
+$ kubectl describe package/publicName.version
+```
+
+You can then view the `.template.spec` of the Package to see if there are any issues with the inputs of the Package. These inputs 
+are eventually used to create the App for the InstalledPackage and can lead to failures.
+
+### Debugging PackageRepository CRs
+
+Failures for PackageRepositories can be directly viewed via a PackageRepository's status via the `usefulErrorMessage` property of
+the status. This `usefulErrorMessage` property comes from an App CR that is created as a result of creating a PackageRepository. More 
+information on interpretting the error message from `usefulErrorMessage` can be found under the [Debugging App CRs](#debugging-app-crs).
+The underlying App CR will have the same name as the PackageRepository that you create.
+
+Failures for PackageRepositories most often happen from fetching the PackageRepository contents (i.e. the `.spec.fetch` portion of 
```

Comment:
Can these two sentences be switched in order or combined? e.g. Because the primary responsibility of the PackageRepository is to fetch configuration for all of its Packages, which are typically pulled from an OCI registry as an [imgpkgBundle](/imgpkg/docs/latest/resources/#bundle), failures most often occur in the fetch portion of the PackageRepository spec.

Reply:
I have rewrote everything as follows:

> 
The primary responsibility of the PackageRepository is to fetch the configuration for all of its Packages, which are typically pulled from an OCI registry as an [imgpkgBundle](/imgpkg/docs/latest/resources/#bundle). Failures for PackageRepositories most often happen from fetching the PackageRepository contents (i.e. the `.spec.fetch` portion of the PackageRepository spec).

Failures for PackageRepositories can be viewed directly via the `usefulErrorMessage` property of the PackageRepository's status. This `usefulErrorMessage` property comes from an App CR that is created as a result of creating a PackageRepository. This App is currently always created in the `kapp-controller` namespace, but this may change in the futrue. More information on interpreting the error message from `usefulErrorMessage` can be found under the [Debugging App CRs](#debugging-app-crs). The underlying App CR will have the same name as the PackageRepository that you create.